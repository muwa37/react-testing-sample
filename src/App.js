import { Link, Route, Routes } from 'react-router-dom';

import About from './pages/About';
import Main from './pages/Main';

function App() {
  return (
    <div>
      <Link to='/' data-testid='main-link'>
        main
      </Link>
      <Link to='/about' data-testid='about-link'>
        about
      </Link>
      <Routes>
        <Route path='/' element={<Main />} />
        <Route path='/about' element={<About />} />
      </Routes>
    </div>
  );
}

export default App;
