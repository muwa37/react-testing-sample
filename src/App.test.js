import { fireEvent, render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import App from './App';

describe('test app', () => {
  test('sample', () => {
    render(<App />);

    const helloWorldEl = screen.getByText(/hello world/i);
    const btnEl = screen.getByRole('button');
    const inputEl = screen.getByPlaceholderText(/type in me.../i);

    expect(helloWorldEl).toBeInTheDocument();
    expect(btnEl).toBeInTheDocument();
    expect(inputEl).toMatchSnapshot();
  });

  test('sample async and styles', async () => {
    render(<App />);
    const asyncDataEl = await screen.findByText(/data/i);
    expect(asyncDataEl).toBeInTheDocument();
    expect(asyncDataEl).toHaveStyle({ color: 'red' });
  });

  test('sample testid with event', () => {
    render(<App />);
    const btnByTestId = screen.getByTestId('toggle-btn');
    expect(screen.queryByTestId('toggle-elem')).toBeNull();
    fireEvent.click(btnByTestId);
    expect(screen.getByTestId('toggle-elem')).toBeInTheDocument();
    fireEvent.click(btnByTestId);
    expect(screen.queryByTestId('toggle-elem')).toBeNull();
  });

  test('sample input event', () => {
    render(<App />);
    const inputEl = screen.getByPlaceholderText(/type in me.../i);
    expect(screen.queryByTestId('value-elem')).toContainHTML('');
    userEvent.type(inputEl, 'test');
    expect(screen.queryByTestId('value-elem')).toContainHTML('test');
  });
});
