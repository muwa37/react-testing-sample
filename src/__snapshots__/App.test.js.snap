// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test app sample 1`] = `
<input
  placeholder="type in me..."
  type="text"
/>
`;
